acvm-repo/brillig/src/opcodes.rs

```diff
@@ -111,8 +132,13 @@ pub enum BrilligOpcode {
         function: String,
         /// Destination registers (may be single values or memory pointers).
         destinations: Vec<RegisterOrMemory>,
+        /// Destination value types
+        destination_value_types: Vec<HeapValueType>,
```

Comment:
It isn't entirely clear to me as to why these types need to be attached. Can we not determine this from the `ForeignCallParam`?

Reply:
If I understand correctly, `ForeignCallParam` is the type to hold the value at runtime of the inputs and outputs for the foreign call. It doesn't belong in the opcode representation of the Brillig program, and it doesn't really have typing information. Maybe I'm missing something?

Reply:
I see now you are mitigating that `RegisterOrMemory` does not containing enough information about whether the internal pointer is itself referencing memory. Could you add a Noir test that shows nested arrays as inputs to a foreign call (can be just a new temporary foreign call or accurate printing of nested arrays/slices).

Reply:
> I see now you are mitigating that `RegisterOrMemory` does not containing enough information about whether the internal pointer is itself referencing memory.

Yes, exactly. That info is not available at runtime otherwise.

> Could you add a Noir test that shows nested arrays as inputs to a foreign call (can be just a new temporary foreign call or accurate printing of nested arrays/slices).

Sure! I can add new cases to the `debug_logs` Noir test. I think that would be best.